src/components/SessionList.js

```diff
@@ -3,49 +3,79 @@ import PropTypes from 'prop-types';
 import Session from './Session';
 
 /**
- * eg finding new sessions and removing old
  * The list the sessions live within, responsible for updating the list
+ * eg finding new sessions and removing old
  */
 class SessionList extends React.Component {
   constructor(props) {
     super(props);
     this.state = { instances: [] };
+    this.onInstancesReceived = this.onInstancesReceived.bind(this);
+    this.onPlayerAdded = this.onPlayerAdded.bind(this);
+    this.onInstanceCreated = this.onInstanceCreated.bind(this);
     //this.state = { instances: this.props.testSessions};
   }
 
   componentDidMount() {
     this.updateList();
   }
 
+  onInstancesReceived(err, result) {
+    if (!err) {
+      this.setState({ instances: result });
+    } else {
+      // TODO: handle error
```

Comment:
Can this be handled now or is it dependent on working session disconnects?

Reply:
This is more dependent on the UI of the controller to be finished. So we need to render some for of error.

It will be called when the deepstream server isn't online or if the service is not online.